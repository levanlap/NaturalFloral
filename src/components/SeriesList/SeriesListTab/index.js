import React from 'react'
import { View } from 'react-native'
import { createNavigator, createNavigationContainer, TabRouter } from 'react-navigation'
import {
  Container, Header,
  Left, Right, Body,
  Button, Text
} from 'native-base'
import { Svg } from 'expo'
import TabView from './components/TabView'
import SeriesList from './components/SeriesList'

const { Rect } = Svg

const Menu = () => (
  <Svg width={20} height={16} viewBox='0 0 20 16'>
    <Rect
      fill='#FFFFFF' width='20' height='4'
      rx='2' ry='2'
    />
    <Rect
      fill='#FFFFFF' y='6' width='15'
      height='4' rx='2' ry='2'
    />
    <Rect
      fill='#FFFFFF' y='12' width='10'
      height='4' rx='2' ry='2'
    />
  </Svg>
)

export default function SeriesListTab({ openMenu, navigation, game }) {
  const tabs = {}
  game.seriesTypes.map((seriesType) => {
    const extraProps = { game, seriesType, stackNavigation: navigation }
    tabs[seriesType.title] = {
      screen: (props) => (<SeriesList {...props} {...extraProps} />)
    }
  })

  const Tabs = createNavigationContainer(createNavigator(TabRouter(tabs))(TabView)) // eslint-disable-line

  return (
    <Container style={{ backgroundColor: '#E8E7E7' }}>
      <Header style={{ backgroundColor: '#388E3C' }}>
        <Left style={{ flex: 0.2 }}>
          <Button transparent onPress={openMenu}>
            <Menu />
          </Button>
        </Left>
        <Body>
          <Text style={{ fontFamily: 'Roboto_medium', color: '#FFFFFF' }}>{game.name}</Text>
        </Body>
        <Right style={{ flex: 0.2 }} />
      </Header>
      <View style={{ flex: 1 }}>
        <Tabs />
      </View>
    </Container>
  )
}
