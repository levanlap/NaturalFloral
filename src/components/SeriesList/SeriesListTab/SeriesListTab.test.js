import React from 'react'
import SeriesListTab from './index'

import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const rendered = renderer.create(<SeriesListTab game={{ seriesTypes: [{}] }} />).toJSON()
  expect(rendered).toBeTruthy()
})

it('render correctly', async () => {
  const tree = renderer.create(<SeriesListTab game={{ seriesTypes: [{}] }} />).toJSON()
  expect(tree).toMatchSnapshot()
})
