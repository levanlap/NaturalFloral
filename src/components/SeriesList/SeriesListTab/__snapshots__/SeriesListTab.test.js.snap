// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#E8E7E7",
      },
    ]
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#92C7A9",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 15,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#388E3C",
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            },
            Object {
              "flex": 0.2,
            },
          ]
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <RNSVGSvgView
            align="xMidYMid"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
                false,
                Object {
                  "flex": 0,
                  "height": 16,
                  "width": 20,
                },
              ]
            }
            vbHeight={16}
            vbWidth={20}
          >
            <RNSVGRect
              fill={
                Array [
                  0,
                  1,
                  1,
                  1,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              height="4"
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              rx="2"
              ry="2"
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth="1"
              width="20"
              x="0"
              y="0"
            />
            <RNSVGRect
              fill={
                Array [
                  0,
                  1,
                  1,
                  1,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              height="4"
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              rx="2"
              ry="2"
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth="1"
              width="15"
              x="0"
              y="6"
            />
            <RNSVGRect
              fill={
                Array [
                  0,
                  1,
                  1,
                  1,
                  1,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              height="4"
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={
                Array [
                  "fill",
                ]
              }
              rotation={0}
              rx="2"
              ry="2"
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth="1"
              width="10"
              x="0"
              y="12"
            />
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Roboto_medium",
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            },
            Object {
              "flex": 0.2,
            },
          ]
        }
        virtual={undefined}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 28,
            "margin": 10,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#388E3C",
              "borderBottomLeftRadius": 4,
              "borderColor": "#388E3C",
              "borderTopLeftRadius": 4,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "color": "#388E3C",
                  "fontFamily": "Roboto_medium",
                  "fontSize": 12,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            undefined
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          data={undefined}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          invertStickyHeaders={undefined}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReached={[Function]}
          onEndReachedThreshold={0.8}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          refreshControl={
            <RefreshControlMock
              onRefresh={[Function]}
              refreshing={true}
              tintColor="#388E3C"
              title="Pull to refresh"
              titleColor="#959595"
            />
          }
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "height": "100%",
            }
          }
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <RCTRefreshControl />
          <View />
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
